offer-manager-app/src/main/groovy/life/qbic/portal/offermanager/DependencyManager.groovy

```diff
@@ -333,9 +337,12 @@ class DependencyManager {
     private SearchAffiliationView createSearchAffiliationView() {
         ResourcesService<Affiliation> affiliationResourcesService = this.affiliationService
         SearchAffiliationViewModel searchAffiliationViewModel = new SearchAffiliationViewModel(affiliationResourcesService)
-
+        ArchiveAffiliationController archiveAffiliationController = new ArchiveAffiliationController()
         UpdateAffiliationView updateAffiliationView = createUpdateAffiliationView()
-        SearchAffiliationView searchAffiliationView = new SearchAffiliationView(searchAffiliationViewModel, updateAffiliationView)
+        SearchAffiliationView searchAffiliationView = new SearchAffiliationView(searchAffiliationViewModel, updateAffiliationView, archiveAffiliationController)
+        ArchiveAffiliationOutput archiveAffiliationOutput = new ArchiveAffiliationPresenter(viewModel, searchAffiliationViewModel, searchAffiliationView, affiliationResourcesService)
+        ArchiveAffiliation archiveAffiliation = new ArchiveAffiliation(archiveAffiliationOutput, updateAffiliationDataSource)
+        archiveAffiliationController.setUseCaseInput(archiveAffiliation)
```

Comment:
You could move this to a method that returns an archiveAffiliation usecase such as done for updateAffiliationView (-> createUpdateAffiliationView())

Reply:
You dont like the suggestion @sven1103 ?

Reply:
solved in an in person conversation